### PR TITLE
Support smooth scrolling

### PR DESCRIPTION
**Pull request by szbergeron**
Adds a stub to allow directly pulling smooth scroll events from reason-sdl2

### CODE REVIEW

**Comment by bryphe** [comment]
Awesome! Great to see this making its way in - thanks @szbergeron !

**Comment by szbergeron** [comment]
Depends on changes in https://github.com/revery-ui/esy-sdl2/pull/6, https://github.com/revery-ui/reason-sdl2/pull/25

> Awesome! Great to see this making its way in - thanks @szbergeron !
Thanks, happy to help :smiley:

File: src/Core/Window.re [hunk 1]
```diff
@@ -331,7 +339,9 @@ let _handleEvent = (sdlEvent: Sdl2.Event.t, v: t) => {
   | Sdl2.Event.WindowLeave(_) => Event.dispatch(v.onMouseLeave, ())
   | Sdl2.Event.WindowExposed(_) => Event.dispatch(v.onExposed, ())
   | Sdl2.Event.Quit => ()
-  | _ => ()
+  | _ => 
+    log("Unknown event recieved");
+    ()
```
Comment (szbergeron) [inline src/Core/Window.re]: Prints

File: src/Core/Window.re [hunk 2]
```diff
@@ -280,11 +280,19 @@ let render = (w: t) => {
...
   switch (sdlEvent) {
   | Sdl2.Event.MouseWheel({deltaX, deltaY, _}) =>
+    log("%%% Mousewheel");
     let wheelEvent: Events.mouseWheelEvent = {
       deltaX: float_of_int(deltaX),
       deltaY: float_of_int(deltaY),
     };
     Event.dispatch(v.onMouseWheel, wheelEvent);
+  | Sdl2.Event.MousePan({deltaX, deltaY, _}) =>
+    log("====Pan of " ++ string_of_int(deltaX) ++ ", " ++ string_of_int(deltaY));
```
Comment (szbergeron) [inline src/Core/Window.re]: Prints

**Comment by szbergeron** [comment]
@bryphe is running CI intended to be a privileged action?

**Comment by szbergeron** [comment]
Bunch of funky stuff seems to be going on with CI for this

**Comment by glennsl** [comment]
Hygiene check fails because of formatting. Run `esy format` locally, commit and push to fix. Windows is still working and the others succeeded, but with "issues" that are normal as far as I'm aware.

**Comment by szbergeron** [comment]
Running `esy format` locally doesn't seem to change anything, not sure why the check failed

**Comment by glennsl** [comment]
> Running esy format locally doesn't seem to change anything, not sure why the check failed

Might be different version

**Comment by glennsl** [comment]
This includes a lot of unrelated changes, which makes it really hard to review. Do you think you could try a rebase?

**Comment by szbergeron** [comment]
Oh whoops, figured it would reconcile. I can try rebasing but not sure if that would help.

**Comment by szbergeron** [comment]
Might make more sense to just revert the top two commits and remake them

**Comment by glennsl** [comment]
An interactive rebase allows you to do pretty much that.

**Comment by szbergeron** [comment]
I seem to have gotten myself in a bit of a pickle trying to do it with rebasing, lots of conflicts and none of them make much sense :woman_shrugging:

**Comment by glennsl** [comment]
That happens sometimes. Just `git rebase --abort` and try again.

The way I think about a rebase is that it replays a set of commits onto the HEAD of the branch or ref you specify. Typically the set set of commits is every commit in your branch from the most recent common ancestor, but you can select the commits yourself by doing an interactive rebase with `git rebase -i`.

When you run `git rebase -i`  you'll get a list of commits in your editor. Delete every line with a commit  that should already be in the new base, ie. that isn't yours and aren't merge commits. Save and quit. You'll most likely still get conflicts, but they should hopefully be relatively small and straight forward to fix since it's just your own changes.

Hope that helps! If not, let me know and I'll try to guide you as best I can.

**Comment by glennsl** [comment]
Looks much better! Didn't realize the change was this tiny though. It really would have been much easier to just copy the changes out and back in after resetting. But hopefully you learned something new!

**Comment by glennsl** [comment]
Something's up with the Linux CI:

```
Failed to fetch https://packages.microsoft.com/ubuntu/16.04/prod/dists/xenial/main/binary-amd64/Packages.bz2  Hash Sum mismatch
```

@bryphe?

**Comment by szbergeron** [comment]
@bryphe I'm currently a bit occupied with projects other than libscroll atm, but since scrolling on wayland is currently regressed by the esy-sdl2/reason-sdl2 changes being applied without this one it's probably a good idea to at least wrap up this part so it isn't up in the air. Would you be able to kick the CI to see if that issue was just a transient issue? I don't currently see why it failed (I think the build logs expired maybe?)

**Comment by szbergeron** [comment]
I think this is obsolete now and is anyway superseded by libscroll integration, so closing while I can get libscroll patch at https://github.com/revery-ui/revery/pull/766 working.

**Pull request closed without merging**